File: hub/src/test/unit/java/com/thoughtworks/selenium/grid/hub/ApplicationRegistryTest.java
package com.thoughtworks.selenium.grid.hub;

import static junit.framework.Assert.*;
import org.junit.Test;


public class ApplicationRegistryTest {

    @Test
    public void registryReturnsAValidInstance() {
        assertFalse(null == ApplicationRegistry.registry());
    }

    public void registryReturnsASingletonInstance() {
        assertSame(ApplicationRegistry.registry(), ApplicationRegistry.registry());
    }

    @Test
    public void remoteControlPoolReturnsAValidPool() {
        assertFalse(null == ApplicationRegistry.registry().remoteControlPool());
    }

    @Test
    public void environmentManagerReturnsAUniqueInstance() {
        assertSame(ApplicationRegistry.registry().environmentManager(),
                   ApplicationRegistry.registry().environmentManager());
    }

    @Test
    public void gridConfigurationReturnsAUniqueInstance() {
        assertSame(ApplicationRegistry.registry().gridConfiguration(),
                   ApplicationRegistry.registry().gridConfiguration());
    }

}

File: hub/src/test/unit/com/thoughtworks/selenium/grid/hub/application_registry_test.rb
require File.expand_path(File.dirname(__FILE__) + "/../../../../../unit_test_helper")

include_class 'com.thoughtworks.selenium.grid.hub.ApplicationRegistry'

class EnvironmentTest < Test::Unit::TestCase

  test "registry returns a singleton instance" do
    assert_equal ApplicationRegistry.registry, ApplicationRegistry.registry
  end

end
